portal-impl/src/com/liferay/portal/service/impl/PortletLocalServiceImpl.java

```diff
@@ -822,8 +828,32 @@ public void initEAR(
 					servletContextName, servletContext, xmls[1],
 					servletURLPatterns, pluginPackage));
 
+			Map<String, String> newPortletIdsByStrutsPath = new HashMap<>();
```

Comment:
This piece is weird, we collect the incoming WAR's struts path, just for the duplicated warning? Shouldn't we add the new values to the _portletIdsByStrutsPath map?